### PR TITLE
Add typealiases. Codestyle improvements.

### PR DESCRIPTION
**Pull request by dewarder**
Hello! Thanks a lot for your library. 

The purpose of this PR is to clean up some parts of codebase according to [Coding Conventions](https://kotlinlang.org/docs/reference/coding-conventions.html)
Also, this PR's brings typealiases for all selectors and some other cases which significantly improve readability.

Here is a short list of general changes:
 - Change builder methods to use `apply()` which allows to omit `return this`
 - Replace all lambdas of selectors with its typealiases, e.g. `FlashSelector`, `QualitySelector`
 - Specify return type explicitly for methods which are exposed. This change allows to not break compatibility of open interface in case of any kind of refactoring.
 - Make some methods to use expression body instead of block one
 - Small changes (use range operator, sort modifiers, convert lamda to reference)

If some changes don't fit your needs feel free to comment what should be changed or reverted.

### CODE REVIEW

File: fotoapparat/src/main/java/io/fotoapparat/FotoapparatBuilder.kt [hunk 1]
```diff
@@ -35,127 +31,109 @@ class FotoapparatBuilder internal constructor(private var context: Context) {
     /**
      * @param selector camera sensor position from list of available positions.
      */
-    fun lensPosition(selector: Iterable<LensPosition>.() -> LensPosition?): FotoapparatBuilder {
-        lensPositionSelector = selector
-        return this
-    }
+    fun lensPosition(selector: LensPositionSelector): FotoapparatBuilder =
```
Comment (Diolor) [inline fotoapparat/src/main/java/io/fotoapparat/FotoapparatBuilder.kt]: This style is slightly more readable imho.

```
fun lensPosition(selector: LensPositionSelector): FotoapparatBuilder = apply { 
    lensPositionSelector = selector 
}
```

File: fotoapparat/src/main/java/io/fotoapparat/FotoapparatBuilder.kt [hunk 2]
```diff
@@ -35,127 +31,109 @@ class FotoapparatBuilder internal constructor(private var context: Context) {
...
 
     /**
      * @param scaleType of preview inside the view.
      */
-    fun previewScaleType(scaleType: ScaleType): FotoapparatBuilder {
-        this.scaleType = scaleType
-        return this
-    }
+    fun previewScaleType(scaleType: ScaleType): FotoapparatBuilder =
+            apply { this.scaleType = scaleType }
```
Comment (Diolor) [inline fotoapparat/src/main/java/io/fotoapparat/FotoapparatBuilder.kt]: same as l35

Comment (Diolor) [reply fotoapparat/src/main/java/io/fotoapparat/FotoapparatBuilder.kt]: there are 3 more in the class

File: fotoapparat/src/main/java/io/fotoapparat/capability/provide/CapabilitiesProvider.kt [hunk 3]
```diff
@@ -21,7 +21,7 @@ private fun SupportedParameters.getCapabilities(): Capabilities {
             canSmoothZoom = supportedSmoothZoom,
             maxMeteringAreas = maxNumMeteringAreas,
             jpegQualityRange = jpegQualityRange,
-            antiBandingModes = supportedAutoBandingModes.extract { it.toAntiBandingMode() },
+            antiBandingModes = supportedAutoBandingModes.extract(String::toAntiBandingMode),
```
Comment (Diolor) [inline fotoapparat/src/main/java/io/fotoapparat/capability/provide/CapabilitiesProvider.kt]: I am a bit sceptical about the `::` pattern (there are few in other files too) but I don't have hard feelings

File: fotoapparat/src/main/java/io/fotoapparat/log/FileLogger.kt [hunk 4]
```diff
@@ -9,9 +9,7 @@ import java.io.IOException
  */
 class FileLogger(private val file: File) : Logger {
 
-    private val writer: FileWriter by lazy {
-        FileWriter(file)
-    }
+    private val writer: FileWriter by lazy { FileWriter(file) }
```
Comment (Diolor) [inline fotoapparat/src/main/java/io/fotoapparat/log/FileLogger.kt]: also here, shorter != more readable

**Comment by dewarder** [comment]
Thanks for merging PR! Regarding your comments I've just tried to follow conventions mentioned above. And in the end such small things can vary from developer to developer.
So in case if something related to this PR should be changed let me know I will try to help with it.

**Pull request merged**